### DISCUSSION TITLE
Unicode support in MSSQL database (varchar fields instead of nvarchar fields)

### DISCUSSION THREAD

**Discussion by tobias-graf-p** [requarks/wiki#3594] [Error / Bug Report]
We use wiki.js 2.x (latest version) with a MSSQL 2014 (v12.0.6118.4) database and SQL_Latin1_General_CP1_CI_AS as the database collation. When pasting Unicode characters into a wiki page, it gets displayed in the UI (preview of the markdown editor), but on saving, the Unicode characters get replaced by "?" or "??". We tried for example this Unicode character: 🥧 

In the database, we noticed, that most textfields are of type "nvarchar" or "ntext", which support Unicode characters. But the following fields are of type "varchar", which does not support Unicode characters:
- dbo.pages.content
- dbo.pages.render
- dbo.pageHistory.content

We would love to see wiki.js support unicode characters on wiki pages. Do we miss something to achieve that? Or could this be changes in a future release?

**Comment by ek68794998** [comment]
Can confirm that updating the columns that Tobias mentioned to `nvarchar(MAX)` fixed the issue for me.

    ALTER TABLE dbo.pages
    ALTER COLUMN content NVARCHAR(MAX) NULL

This is probably a fairly trivial fix to implement when creating the table structure, but for existing users they may have to edit the column by hand.

**Reply by sfedor** [reply]
The same issue is with `text` fields. For example, I had to change the type of `pages.toc` field to `ntext`.
